core/src/server/rpc_module.rs

```diff
@@ -80,19 +71,45 @@ pub type ConnectionId = usize;
 /// Max response size.
 pub type MaxResponseSize = usize;
 
-/// Raw response from an RPC
-/// A 3-tuple containing:
-///   - Call result as a `String`,
-///   - a [`mpsc::UnboundedReceiver<String>`] to receive future subscription results
-///   - a [`crate::server::helpers::SubscriptionPermit`] to allow subscribers to notify their [`SubscriptionSink`] when they disconnect.
-pub type RawRpcResponse = (MethodResponse, mpsc::UnboundedReceiver<String>, SubscriptionPermit);
+/// Error that may occur during `SubscriptionSink::try_send`.
+#[derive(Debug)]
+pub enum TrySendError {
+	/// The channel is closed.
+	Closed(SubscriptionMessage),
+	/// The channel is full.
+	Full(SubscriptionMessage),
+}
+
+#[derive(Debug, Clone)]
+/// Represents whether a subscription was answered or not.
+pub enum SubscriptionAnswered {
```

Comment:
I think we don't need `MethodResponse` for the subscriptions but I kept it for now to all calls to return MethodResponse.

Reply:
nvm, it's required by the Logger so no way around it right now.